File: finished/find_missing_observations.py
# https://leetcode.com/problems/find-missing-observations/

from typing import List


class Solution:
    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
      m = len(rolls)
      npm = n + m
      total = npm * mean

      msum = sum(rolls)
      remaining = total - msum
      if remaining > 6 * n or remaining < 1 * n:
        return []

      navg = remaining / n
      navg_low = int(navg // 1)
      low_sum = navg_low * n
      n_missing = remaining - low_sum

      res = []
      for i in range(n-n_missing):
        res.append(navg_low)
      for i in range(n_missing):
        res.append(navg_low + 1)
      return res

File: src/go/finished/find_missing_observations.go
package leetcode

import "math"

// https://leetcode.com/problems/find-missing-observations/

func missingRolls(rolls []int, mean int, n int) []int {
	totalRolls := len(rolls) + n
	sum := mean * totalRolls
	remaining := sum
	for _, r := range rolls {
		remaining -= r
	}

	if remaining > 6*n || remaining < 1*n {
		return []int{}
	}

	navg := float64(remaining) / float64(n)
	nLow := math.Floor(navg)
	nLowInt := int(nLow)
	lowSum := nLow * float64(n)
	leftOver := float64(remaining) - lowSum
	leftOverInt := int(leftOver)

	res := []int{}
	for i := 0; i < (n - leftOverInt); i++ {
		res = append(res, nLowInt)
	}
	for i := 0; i < leftOverInt; i++ {
		res = append(res, nLowInt+1)
	}
	return res
}
